src/Domain/DoctrineDonationRepository.php

```diff
@@ -156,11 +155,13 @@ public function allocateMatchFunds(Donation $donation): string
             $amountNewlyMatched = bcadd($amountNewlyMatched, $newWithdrawalAmount, 2);
         }
 
+        if (count($newWithdrawals) > 0) {
+            $this->getEntityManager()->flush();
+        }
+
```

Comment:
I'm not sure if this will give us any appreciable speedup, since `\Doctrine\ORM\UnitOfWork::commit` already looks like it returns without doing any DB work if there's nothing to update:

```
        if (
            ! ($this->entityInsertions ||
                $this->entityDeletions ||
                $this->entityUpdates ||
                $this->collectionUpdates ||
                $this->collectionDeletions ||
                $this->orphanRemovals)
        ) {
            $this->dispatchOnFlushEvent();
            $this->dispatchPostFlushEvent();

            $this->postCommitCleanup($entity);

            return; // Nothing to do.
        }
```

Reply:
@bdsl That sounds very likely right. It's avoiding the explicit, extraneous transaction in `Adapter::saveFundingsToDatabase()` which I expect to make the main difference.

Reply:
OK - IMHO unless there's a reason you think making this flush conditional might speed things up or you want to test how it performs then I think we should keep it unconditional for simplicity.

Reply:
I think which is clearer is debatable but reverted + added a comment to help understand which entities are involved when quickly scanning the code, in bc8cfdb042530c06e4da5a5f8df5ea1b6867ff58